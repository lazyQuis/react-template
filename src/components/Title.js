import React, { PropTypes, Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { trace } from 'actions/trace';
import { BaseComponent } from 'react-libs';
import { global } from 'constants/config';

export class Title extends BaseComponent {
  constructor(props) {
    super();
    this.click = this.click.bind(this);
  }

  click() {
    if(this.props.router) this.props.router.push(this.props.jump);
  }

  componentDidMount() {
    this.props.trace(this.props.text);
  }

  componentDidUpdate(prevProps, prevState) {
    this.props.trace(this.props.text);
  }

  render() {
    const { text, jump } = this.props;


    return (
      <div className="scene__title">
        {global.mode} page is <span style={{color:'red'}}>{text}</span>.
        { jump ? <button onClick={this.click} className="scene__button">{jump}</button> : null }
      </div>
    );
  };
};

Title.propTypes = {
  jump  : PropTypes.string,
  router: PropTypes.object,
  text  : PropTypes.string
};

Title.defaultProps = {
  jump   : '',
  text   : ''
};

export default connect(
    state => ({
      // log: state.trace.log
    }),
    dispatch => bindActionCreators({
      trace
    }, dispatch)
)(Title);
