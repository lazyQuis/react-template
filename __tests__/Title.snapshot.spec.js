import React from 'react'
import renderer from 'react-test-renderer'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import Title from '@/components/Title.js'

const routes =[
  {
    path     : '',
    component: '',
    routes   : [
      {
        tagName  : 'Home : Page',
        path     : `/home/page`,
        component: '',
        routes   : [
          {
            tagName  : 'Info',
            path     : `/home/page/info`,
            component: ''
          }, {
            path     : `/home/page/`,
            component: ''
          }
        ]
      }
    ]
  }
]
const store = createStore(
  () => ({
    sys:{
      info: {message: 'Info Message'}
    },
    routeData: routes
  })
)

describe('Title.js', () => {
  it('<Title />', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Title />
      </Provider>
    )
    let snapshot = component.toJSON()
    expect(snapshot).toMatchSnapshot()
  })

  it('<Title jump="/home/page" history={[]} pathname="/home/page/info" />', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Title jump="/home/page" history={[]} pathname="/home/page/info" />
      </Provider>
    )
    let snapshot = component.toJSON()
    expect(snapshot).toMatchSnapshot()
  })
})
