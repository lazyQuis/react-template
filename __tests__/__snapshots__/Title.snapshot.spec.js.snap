// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title.js <Title /> 1`] = `
<div
  className="scene__title"
>
  production
  <br />
  <button
    className="scene__button"
    onClick={[Function]}
  >
    MultiAction API
  </button>
  <div
    className="scene__msg"
  >
    Info Message
  </div>
</div>
`;

exports[`Title.js <Title jump="/home/page" history={[]} pathname="/home/page/info" /> 1`] = `
<div
  className="scene__title"
>
  production
  <br />
  <div
    className="crumbs"
  >
    <ol>
      <li
        className="crumbs__item"
      >
        <p
          className="crumbs__link"
          onClick={[Function]}
        >
          Home : Page
        </p>
        <i
          className="material-icons crumbs__arrow"
        >
          îŒ•
        </i>
      </li>
      <li
        className="crumbs__item"
      >
        <p
          className="crumbs__link crumbs__link--strong"
          onClick={[Function]}
        >
          Info
        </p>
      </li>
    </ol>
  </div>
  <button
    className="scene__button"
    onClick={[Function]}
  >
    Next Page
  </button>
  <button
    className="scene__button"
    onClick={[Function]}
  >
    MultiAction API
  </button>
  <div
    className="scene__msg"
  >
    Info Message
  </div>
</div>
`;
